base/meta.jl

```diff
@@ -18,6 +28,33 @@ isexpr(ex,       head)          = false
 
 isexpr(ex,       head, n::Int)  = isexpr(ex, head) && length(ex.args) == n
 
+isline(ex) = isexpr(ex, :line) || isa(ex, LineNumberNode)
+
+rmlines(xs) = filter(x->!isline(x), xs)
+rmlines(x::Expr) = Expr(x.head, rmlines(x.args)...)
+
+function unblock(ex)
+    isexpr(ex, :block) || return ex
+    exs = filter(ex->!isline(ex), ex.args)
+    length(exs) == 1 || return ex
+    return unblock(exs[1])
+end
+
+namify(ex::Expr) = isexpr(ex, :.) ? ex : namify(ex.args[1])
+namify(ex::QuoteNode) = ex.value
+namify(sy::Symbol) = sy
+
+uncurly(ex) = isexpr(ex, :curly) ? ex.args[1] : ex
+
+isdef(ex) = isexpr(ex, :function) || (isexpr(ex, :(=)) && isexpr(ex.args[1], :call))
+
+subs(ex::Expr, s, s′) = ex == s ? s′ : Expr(ex.head, map(ex -> subs(ex, s, s′), ex.args)...)
+
+subs(ex, s, s′) = ex == s ? s′ : ex
+
+macro expand(ex)
+      :(macroexpand($(Expr(:quote, ex))))
```

Comment:
``quot(ex)`` instead of ``Expr(:quote, ex)`` here? A few too many indents as well.